funcx_sdk/funcx/serialize/facade.py

```diff
@@ -25,9 +45,59 @@ def __init__(self):
         for key in METHODS_MAP_DATA:
             self.methods_for_data[key] = METHODS_MAP_DATA[key]()
 
+    def _start_off_process_checker(self):
+        """ Kicks off off_process_checker using subprocess and returns the port on which it
+        is listening.
+        """
+        std_file = f'/tmp/{uuid.uuid4()}'
+        logger.debug(f"Writing off_process_checker logs to: {std_file}")
+        std_out = open(std_file, 'w')
+        self.serialize_proc = subprocess.Popen(['off_process_checker.py'], stdout=std_out, stderr=subprocess.STDOUT)
+        with open(std_file) as f:
+            for i in range(200, 3000, 200):
+                time.sleep(i / 100)  # Sleep incrementally waiting for process start
+                for line in f.readlines():
+                    if 'BINDING TO' in line:
+                        _, port = line.strip().split(':')
+                        port = int(port)
+                        return port
+                    elif 'CRITICAL ERROR' in line:
```

Comment:
This does not match what it prints? `f"OFF_PROCESS_CHECKER FAILURE, Exception:{e}"`

Reply:
Fixed.